File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.swift
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     var val: Int
 *     var left: TreeNode?
 *     var right: TreeNode?
 *     init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */

class Solution {
    func maxDepth(_ root: TreeNode?) -> Int {
        guard let root = root else{
            // no root - 0 depth
            return 0
        }
        
        // we get the max depth of the right and left subrees
        var maxRight = maxDepth(root.right)
        var maxLeft = maxDepth(root.left)
        
        // add root 
        return max(maxLeft, maxRight) + 1
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque 

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        result = [] 
        # no root - empty list 
        if not root: 
            return result

        queue = deque() 

        # add root 
        if root: 
            queue.append(root)

        while len(queue) > 0:
            level = []
            for i in range(len(queue)):
                current = queue.popleft()
                level.append(current.val)
                if current.left:
                    # add to queue
                    queue.append(current.left)
                if current.right:
                    queue.append(current.right)
            # append to result
            result.append(level)
        return result 



        